zha/application/platforms/light/__init__.py

```diff
@@ -1062,7 +1063,7 @@ def restore_external_state_attributes(
         xy_color: tuple[float, float] | None,
         hs_color: tuple[float, float] | None,
         color_mode: ColorMode | None,
-        effect: str | None,
+        effect: str,
```

Comment:
`restore_external_state_attributes` can still get `None` from HA, right? E.g. if the light is off or if the `core.restore_state` JSON file is deleted from HA(?)

I think we should revert this change and make sure that `effect is not None` before restoring it below (currently line 1087).

---

"Why should we not restore `None` effect? anymore"
Well, it'll now behave similar to all other state attributes. `None` means "no data" instead of "no effect" now. So, we should follow the same behavior as we do for the other state attributes.
For those, we do not restore `None` because some state attributes may already be populated from the zigpy cache, though it may be slightly outdated for non-attribute reporting lights.